src/librbd/object_map/Request.cc

```diff
@@ -22,7 +22,10 @@ bool Request::should_complete(int r) {
   switch (m_state)
   {
   case STATE_REQUEST:
-    if (r < 0) {
+    if (r == -ETIMEDOUT &&
```

Comment:
@dillaman On the weekend, the update object map timeout occurred (there are slow request) when we were writing in 1M seq. In this corner-case, librbd does nothing but return IO errors to APPs, let it be handled by the APPs. Please help to take a look. Thanks.
```
2020-03-28 18:21:54.551307 7f5be97fa700 20 librbd::object_map::UpdateRequest: 0x7f5bdc047490 update_object_map: ictx=0x2b5f7e0, oid=rbd_object_map.14d156b8b4567, [24915,24916) = ->1
......
2020-03-28 18:22:25.551554 7f5be9ffb700 20 librbd::object_map::UpdateRequest: 0x7f5bdc047490 handle_update_object_map: r=-110
2020-03-28 18:22:25.551563 7f5be9ffb700 20 librbd::object_map::UpdateRequest: 0x7f5bdc047490 update_in_memory_object_map:
2020-03-28 18:22:25.551569 7f5be9ffb700 20 librbd::object_map::Request: 0x7f5bdc047490 should_complete: r=-110
2020-03-28 18:22:25.551575 7f5be9ffb700 -1 librbd::object_map::Request: failed to update object map: (110) Connection timed out
2020-03-28 18:22:25.551637 7f5be9ffb700 -1 librbd::object_map::InvalidateRequest: 0x7f5be00234c0 invalidating object map in-memory
2020-03-28 18:22:25.551643 7f5be9ffb700 -1 librbd::object_map::InvalidateRequest: 0x7f5be00234c0 invalidating object map on-disk
2020-03-28 18:22:25.569080 7f5be9ffb700 -1 librbd::object_map::InvalidateRequest: 0x7f5be00234c0 should_complete: r=0
2020-03-28 18:22:25.569111 7f5be9ffb700 20 librbd::object_map::Request: 0x7f5bdc047490 should_complete: r=0
2020-03-28 18:22:25.569113 7f5be9ffb700 20 librbd::object_map::Request: INVALIDATE
2020-03-28 18:22:25.569117 7f5be9ffb700 20 librbd::ObjectMap: 0x7f5bdc003820 handle_detained_aio_update: cell=0x7f5bdc172ff0, r=0
```

Reply:
You should return `true` here and not invoke `finish_request` below. I also wonder if the "cure" is worse than the original problem since now your iSCSI clients will receive IO errors.

Reply:
Yup, I think iSCSI will handles this IO errors case. Updated, thank you.

Reply:
Yeah, by bubbling errors back to the clients and having them report IO failures.

Reply:
Yes, in this case, we set `rados_osd_op_timeout = 31` (`osd_heartbeat_grace` + `osd_heartbeat_interval` + 5).